R/check_authorized_users.R

```diff
@@ -0,0 +1,78 @@
+#' Check if a user is authorized to modify model
+#' directories. This function verifies whether a GitHub
+#' user is authorized to modify specific model directories
+#' in a Hub by checking the designated users in model
+#' metadata.
+#'
+#' @param changed_dirs Character vector. Directory names
+#' that have been changed.
+#' @param gh_actor Character. GitHub username of the person
+#' making changes.
+#' @param base_hub_path Character. Path to the base hub
+#' directory.
+#' @param write_status Logical. Whether to write a status
+#' file on success (default: TRUE).
+#'
+#' @return NULL invisibly. Exits with error if the user is
+#' unauthorized and prints success message if authorized.
+#'
+#' @export
+check_authorized_users <- function(
+  changed_dirs,
+  gh_actor,
+  base_hub_path
+) {
+  checkmate::assert_character(changed_dirs, min.len = 1)
+  checkmate::assert_scalar(gh_actor)
+  checkmate::assert_string(base_hub_path)
+
+  model_metadata <- hubData::load_model_metadata(base_hub_path)
+
+  if (nrow(model_metadata) == 0) {
+    cli::cli_abort(
+      "Error: Could not load model metadata from {base_hub_path}!"
+    )
+  }
```

Comment:
`hubData::load_model_metadata` has its own [error handling](https://github.com/hubverse-org/hubData/blob/0e6150f8dd84e39e2b3add18f6d4b14d1fca958b/R/load_model_metadata.R#L32-L55). Defer to that.
 ```suggestion
  if (nrow(model_metadata) == 0) {
    cli::cli_abort(
      "Error: Could not load model metadata from {base_hub_path}!"
    )
  }
```